### PR TITLE
Include Windows and macOS Targets for CI

### PR DESCRIPTION
**Pull request by tareknaser**
## Description
We found that certain bugs appear in macOS but not in Linux, which was hard to spot since we only tested on Linux. For example, system time representation can differ across operating systems.
This PR fixes it by adding Windows and macOS CI jobs for each commit/PR. 
## Changes
The change wasn't straightforward because the build initially failed on Windows and macOS due to:
- Compatibility issues with Windows for `pprof` dependency
- A failed build script for the `data-pdf` crate on macOS, causing the entire workspace build to fail

This PR resolves the issues by:
- Removing the dependency on `pprof` to fix the Windows build
- Correcting the path of `libpdfium` in the build script for the `data-pdf` crate

Currently, we only run CI tests for macOS Intel because using macOS ARM requires payment for GitHub Actions. See more information in [macOS runner pricing](https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/#new-macos-runner-pricing)

For now, we've added support for macOS Intel and included a GitHub Actions workflow for macOS ARM. Just as a placeholder within the release job, which is not yet implemented, with a "FIXME". 
It may be reasonable to run macOS ARM tests exclusively for release builds.

Other changes made include adding the logo to the README and including badges for CI and the license.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: macos-13-xlarge
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+      - name: Run tests
+        run: cargo test --workspace --verbose --release
+
+        # FIXME: Upload Binary step
```
[the remark this answers is not in the document]
Comment (tareknaser) [reply .github/workflows/release.yml:26]: Great. 


Would you like to enable Mac ARM CI workflows (paid) for each push or pull request to the main branch, or only for the release workflow?

[the remark this answers is not in the document]
Comment (kirillt) [reply .github/workflows/release.yml:26]: Initially we've discussed running it only for release workflow, but actually I think each push into `main` branch would be also fine. Not pull request though.

[the remark this answers is not in the document]
Comment (tareknaser) [reply .github/workflows/release.yml:26]: Okay, let's set up the workflow to run on each push to `main` for now. We'll add it to the release workflow once it's in place.

**Comment by tareknaser** [comment]
I've set up a new workflow to run under the following conditions:
- Whenever there's a push to the main branch
- Weekly (Should we switch to running monthly instead?)

This workflow runs tests on different platforms, including Linux, Windows, and macOS (ARM and Intel).

**Review by kirillt — APPROVED** [review]

**Pull request merged**